sitemedia/js/controllers/ittpanel_controller.js

```diff
@@ -28,5 +46,86 @@ export default class extends Controller {
             }
         }
+
+        if (this.isDesktop()) {
+            if (this.transcriptionAndTranslationOpen()) {
+                // when transcription and translation are both opened, align their contents line-by-line
+                this.alignLines();
+            } else {
+                // when one of those two toggles is closed, remove inline styles from lines (alignment no longer needed)
+                this.transcriptionTarget
+                    .querySelectorAll("li")
+                    .forEach((li) => {
+                        li.removeAttribute("style");
+                    });
+                this.translationTarget.querySelectorAll("li").forEach((li) => {
+                    li.removeAttribute("style");
+                });
+            }
+        }
+    }
+
+    transcriptionAndTranslationOpen() {
+        // check toggle targets to find out whether both transcription and translated are open
+        return ["transcription-on", "translation-on"].every(
+            (id) =>
+                this.toggleTargets.find((target) => target.id === id)
+                    .checked === true
+        );
+    }
+
+    alignLines() {
+        // align each line of transcription to translation
+        const edLines = this.transcriptionTarget.querySelectorAll("li");
+        const trLines = this.translationTarget.querySelectorAll("li");
+        // only align as many lines as we need to
+        const [lessLines, moreLines] =
+            edLines.length < trLines.length
+                ? [trLines, edLines]
+                : [edLines, trLines];
+        lessLines.forEach((li, i) => {
+            if (li && moreLines[i]) {
+                // compare top of lines by position in list
+                const thisLiPos = li.getBoundingClientRect();
+                const thatLiPos = moreLines[i].getBoundingClientRect();
+                // add padding-top to the elements that need it
+                if (thisLiPos.top < thatLiPos.top) {
+                    li.style.paddingTop = `${thatLiPos.top - thisLiPos.top}px`;
+                } else if (thisLiPos.top > thatLiPos.top) {
+                    moreLines[i].style.paddingTop = `${
+                        thisLiPos.top - thatLiPos.top
+                    }px`;
+                }
+            }
+        });
+    }
+
+    clickAnnotation(e) {
+        // on mousedown to edit content, clear any style that was added to li elements.
+        // otherwise, modified HTML will be loaded inside the editor, and saved with the annotation!
+        if (this.isDesktop() && e.target.tagName === "ANNOTATION-BLOCK") {
+            const lines = this.transcriptionTarget.querySelector(
+                "annotation-block"
+            )
+                ? this.transcriptionTarget.querySelectorAll("li")
+                : this.translationTarget.querySelectorAll("li");
+            lines.forEach((li) => {
+                li.removeAttribute("style");
+            });
+        }
```

Comment:
You're right, we do already do this—not in tahqiq, but on saving the annotation in PGP (`Annotation.sanitize_html`). I added this because I was seeing the extra padding persisting in the text editor—I forgot that it would still appear in the text editor, but then as soon as you click "Save" it would be stripped.

I wonder if that would be confusing for users to see in the editor?

![Screenshot 2023-06-08 at 1 48 29 PM](https://github.com/Princeton-CDH/geniza/assets/7234006/3c1e0122-015f-43b9-ae21-d7230063a343)

If so, to me that seems like a good argument for switching to a line-number and data attribute/class based approach. I would need to add that data attribute to the sanitization but that's simple.

Otherwise probably not worth the effort since the padding based on pixels works essentially just as well, even if the concept is slightly different. (Could potentially even just use CSS to set `padding-top` to 0 for all `li` inside the editor element if it's purely a visual problem)

Reply:
I think it might be confusing in the editor, yeah - could imagine users trying to delete that extra space and not understanding why they can't.

If you do it with a class, then you could only apply the style when it's not in the editor.  (Or, it would be automatically ignored anyway because it's in an iframe and tiny editor has its own styles, right?)

Reply:
`sanitize_html` sounds like a much better place to do the cleanup if it's workable

Reply:
@blms probably best to check with the team about how much of a problem this is and how much time they want you to spend revisiting.